sklearn/utils/validation.py

```diff
@@ -26,7 +27,14 @@ class DataConversionWarning(UserWarning):
 
 class NonBLASDotWarning(UserWarning):
     "A warning on implicit dispatch to numpy.dot"
-    pass
+
+
+class NotFittedError(ValueError, AttributeError):
+    """
+    Exception class to raise if estimator is used before fitting
+
+    Inherits from both ValueError and AttributeError to help exception handling
```

Comment:
Ah... thanks for pointing out... @MechCoder had advised me the same in some recent PR of mine... I made the same mistake again... sorry for that... will fix that!